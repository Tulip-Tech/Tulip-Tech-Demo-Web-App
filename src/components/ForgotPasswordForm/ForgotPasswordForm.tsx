import React from 'react';
import { useTranslation } from 'react-i18next';

import styles from './ForgotPasswordForm.module.scss';

import Button from '#components/Button/Button';
import TextField from '#components/TextField/TextField';
import type { FormErrors } from '#types/form';
import type { ForgotPasswordFormData } from '#types/account';
import FormFeedback from '#components/FormFeedback/FormFeedback';
import { testId } from '#src/utils/common';

type Props = {
  onSubmit: React.FormEventHandler<HTMLFormElement>;
  onChange: React.ChangeEventHandler<HTMLInputElement | HTMLTextAreaElement>;
  onBlur: React.FocusEventHandler<HTMLInputElement | HTMLTextAreaElement>;
  error?: string;
  errors: FormErrors<ForgotPasswordFormData>;
  value: ForgotPasswordFormData;
  submitting: boolean;
};

const ForgotPasswordForm: React.FC<Props> = ({ onSubmit, onChange, value, errors, submitting, onBlur }: Props) => {
  const { t } = useTranslation('account');

  return (
    <form onSubmit={onSubmit} data-testid={testId('forgot-password-form')} noValidate className={styles.forgotPasswordForm}>
      <h2 className={styles.title}>{t('reset.forgot_password')}</h2>
      {errors.form ? <FormFeedback variant="error">{errors.form}</FormFeedback> : null}
      <p className={styles.text}>{t('reset.forgot_text')}</p>
      <TextField
        value={value.email}
        onChange={onChange}
        onBlur={onBlur}
        label={t('reset.email')}
        placeholder={t('reset.email')}
        error={!!errors.email || !!errors.form}
        helperText={errors.email}
        required
        name="email"
        type="email"
      />
      <Button type="submit" className={styles.button} fullWidth color="primary" disabled={submitting} label={t('reset.email_me')} />
    </form>
  );
};

export default ForgotPasswordForm;
