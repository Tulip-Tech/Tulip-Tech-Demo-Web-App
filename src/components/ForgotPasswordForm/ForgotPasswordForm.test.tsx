import React from 'react';
import { render } from '@testing-library/react';

import ForgotPasswordForm from './ForgotPasswordForm';

describe('<ForgotPasswordForm>', () => {
  test('renders and matches snapshot type forgot', () => {
    const { container } = render(
      <ForgotPasswordForm submitting={false} onSubmit={vi.fn()} onChange={vi.fn()} value={{ email: '' }} errors={{}} onBlur={vi.fn()} />,
    );

    expect(container).toMatchSnapshot();
  });
});
