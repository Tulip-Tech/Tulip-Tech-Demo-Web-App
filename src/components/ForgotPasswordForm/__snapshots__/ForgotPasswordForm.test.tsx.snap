// Vitest Snapshot v1

exports[`<ForgotPasswordForm> > renders and matches snapshot type forgot 1`] = `
<div>
  <form
    data-testid="forgot-password-form"
    novalidate=""
  >
    <h2
      class="_title_27331a"
    >
      reset.forgot_password
    </h2>
    <p
      class="_text_27331a"
    >
      reset.forgot_text
    </p>
    <div
      class="_textField_e16c1b"
    >
      <label
        class="_label_e16c1b"
        for="text-field_1235_email"
      >
        reset.email
      </label>
      <div
        class="_container_e16c1b"
      >
        <input
          class="_input_e16c1b"
          id="text-field_1235_email"
          name="email"
          placeholder="reset.email"
          required=""
          type="email"
          value=""
        />
      </div>
    </div>
    <button
      aria-disabled="false"
      class="_button_f8f296 _button_27331a _primary_f8f296 _outlined_f8f296 _fullWidth_f8f296"
      type="submit"
    >
      <span>
        reset.email_me
      </span>
    </button>
  </form>
</div>
`;
